src/cpp/session/SessionHttpMethods.cpp

```diff
@@ -533,10 +534,24 @@ void handleConnection(boost::shared_ptr<HttpConnection> ptrConnection,
                                            &hostPageUrl) ;
             if (error)
                LOG_ERROR(error);
-
+            
             // note switch to project
             if (!switchToProject.empty())
             {
+               // if we don't have a project file then create it (can
+               // occur when e.g. opening a project from a directory for
+               // which we don't yet have a .Rproj file)
+               FilePath projFile = module_context::resolveAliasedPath(switchToProject);
```

Comment:
Thanks for catching that -- we need to check against `kProjectNone` here, to be safe.